shell-ui/src/platform/service/k8s.js

```diff
@@ -0,0 +1,113 @@
+//@flow
+import { useState, useEffect } from 'react';
+import { useQuery } from 'react-query';
+import axios from 'axios';
+import { AUTHENTICATED_EVENT } from '../../navbar/events';
+
+const useGetNodesCount = (useQuery: typeof useQuery, k8sUrl: string) => {
+  const navbarElement = document.querySelector('solutions-navbar');
+  const [token, setToken] = useState(null);
+  // $FlowFixMe
+  useQuery('initialToken', () => navbarElement.getIdToken(), {
+    onSucces: (idToken) => setToken(idToken),
+  });
+
+  useEffect(() => {
+    if (!navbarElement) {
+      return;
+    }
+    const onAuthenticated = (evt: Event) => {
+      // $FlowFixMe
+      setToken(evt.detail.id_token);
+    };
+    navbarElement.addEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+    return () =>
+      navbarElement.removeEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+  }, []);
+
+  const queryNodesResult = useQuery(getNodesCountQuery(k8sUrl, token));
+  queryNodesResult.nodesCount = queryNodesResult.data;
+  delete queryNodesResult.data;
+
+  return queryNodesResult;
+};
+
+const useGetVolumesCount = (useQuery: typeof useQuery, k8sUrl: string) => {
+  const navbarElement = document.querySelector('solutions-navbar');
+  const [token, setToken] = useState(null);
+  // $FlowFixMe
+  useQuery('initialToken', () => navbarElement.getIdToken(), {
+    onSucces: (idToken) => setToken(idToken),
+  });
+
+  useEffect(() => {
+    if (!navbarElement) {
+      return;
+    }
+    const onAuthenticated = (evt: Event) => {
+      // $FlowFixMe
+      setToken(evt.detail.id_token);
+    };
+    navbarElement.addEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+    return () =>
+      navbarElement.removeEventListener(AUTHENTICATED_EVENT, onAuthenticated);
+  }, []);
+
+  const queryVolumesResult = useQuery(getVolumesCountQuery(k8sUrl, token));
+  queryVolumesResult.volumesCount = queryVolumesResult.data;
+  delete queryVolumesResult.data;
+
+  return queryVolumesResult;
+};
+
+export const getNodesCountQuery = (
+  k8sUrl: string,
+  token?: string | null,
+): typeof useQuery => {
+  if (!token) {
+    console.error('K8s API Not authenticated');
+    return;
+  }
```

Comment:
IMO it would be better to get rid of this early return in order to have a consistant return type even when the token is not provided. Ensuring the token is provided to perform the request can be done through the `enabled` useQuery option